# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [32]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1    8437.673028         42               высшее             0   
1          1    4024.803754         36              среднее             1   
2          0    5623.422610         33              среднее             1   
3          3    4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0     926.185831         27               высшее             0   
6          0    2879.202052         43               высшее             0   
7          0     152.779569         50              среднее             1   
8          2    6929.865299         35               высшее             0   
9          0    2188.756445         41              среднее             1   
10         2    4171.483647         36               высшее             0   
11         0     792.701887         40              среднее             1   
12         0  365286.622650         65              среднее             1   
13         0    1846.641941         54  неоконченное высшее             2   
14         0    1844.956182         56               высшее             0   
15         1     972.364419         26              среднее             1   
16         0    1719.934226         35              среднее             1   
17         0    2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0   10038.818549         48              среднее             1   
20         1    1311.604166         36              среднее             1   
21         1     253.685166         33              среднее             1   
22         1    1766.644138         24              среднее             1   
23         0     272.981385         21               высшее             0   
24         1  338551.952911         57              среднее             1   
25         0  363548.489348         67              среднее             1   
26         0    2689.137274         41              среднее             1   
27         0     529.191635         28               высшее             0   
28         1     717.274324         26               высшее             0   
29         0  365286.622650         63              среднее             1   
30         1  335581.668515         62              среднее             1   
31         0    1682.083438         47            начальное             3   
32         0    4649.910832         34              среднее             1   
33         0    1548.637544         48              среднее             1   
34         0    4488.067031         35              среднее             1   
35         0  394021.072184         68              среднее             1   
36         0     176.216688         33              среднее             1   
37         0    6448.810860         43               высшее             0   
38         0     597.881827         25               высшее             0   
39         0     650.587796         31               высшее             0   
40         0    1030.362606         30              среднее             1   
41         0    2689.137274         50              среднее             1   
42         0    1257.496190         20  неоконченное высшее             2   
43         0    4375.681384         43  неоконченное высшее             2   
44         1    1362.041728         26              среднее             1   
45         1    1039.451741         49              среднее             1   
46         2    2262.712304         37              среднее             1   
47         1    2689.137274         33               высшее             0   
48         0    3341.067886         45              среднее             1   
49         0    1181.443228         54              среднее             1   

            family_status  family_status_id gender  income_type  debt  \
0  

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

Поскольку мне дали слово и разрешили вести анализ данных, то хочу указать, что в таблице с общим трудовым стажем присутствуют значения, превышающие 100 лет, а так же отрицательные значения. На них стоит обратить внимание-либо в данных допущена ошибка (и, возможно в других столбцах тоже), либо их сейчас ищут Румынские родственники и путешественники во времени. 

Отрицательные значения уже были исправлены ранее, но аномальные значения рабочего стажа (более 100 лет) я решила посчитать, что бы обратить внимание на количество возможных ошибок.

In [22]:
print(data.loc[data['days_employed'] > 36500].count())

children                 3814
days_employed            3814
dob_years                3814
education                3814
education_id             3814
family_status            3814
family_status_id         3814
gender                   3814
income_type              3814
debt                     3814
total_income             3814
purpose                  3814
total_income_category    3814
purpose_category         3814
dtype: int64


3814 значений(!!!), а если смотреть здравым взглядом в разрезе законов, учетом повышающегося стажа в связи с условиями работы и других факторов, этих значений будет еще больше. В данном случае, я бы попросила обратить внимание на этот столбец. Трудовой стаж влияет на многое и нельзя игнорировать ошибки.

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Столбец 'children' указывает нам на уникальные значения количества детей в семье. Ранее мы скорректировали таблицу, удалив все аномальные значения. Таким образом, сейчас в таблице остались значения от 0 до 5 ([1, 0, 3, 2, 4, 5]). Сгруппируем данные в столбце 'children' по данным в столбце 'debt'  и посмотрим на результаты.
С учетом того, что за наличие просрочек по кредиту в столбце отвечает 1, а за их отсутствие 0, то с помощью методов count() и sum(), можно найти общее количество значений по каждому количеству детей и количество просрочек соответственно.

In [23]:
children_debt = data.groupby('children')['debt'].count()    #общее количество по количеству детей
print(children_debt)
children_debt_delay = data.groupby('children')['debt'].sum()     #количество просрочек по каждому количеству детей
children_debt_delay 

children
0    14091
1     4808
2     2052
3      330
4       41
5        9
Name: debt, dtype: int64


children
0    1063
1     444
2     194
3      27
4       4
5       0
Name: debt, dtype: int64

Рассчитаем количество просрочек в процентном соотношении от общих значений по каждому варианту количества детей.

In [24]:
children_debt_delay_percent = children_debt_delay / children_debt
children_debt_delay_percent

children
0    0.075438
1    0.092346
2    0.094542
3    0.081818
4    0.097561
5    0.000000
Name: debt, dtype: float64

По полученным значениям видно, что частота просрочек лежит в диапазоне от 0 до 9,76%. Минимальное значение - 0% соответствует наличию в семье 5 детей, но общее количество таких кредитов - 9, что в случае данной таблицы является нечтожно малым и не может говорить о том, что выдача кредитов семьям с 5 детьми является стратегическо верной, что бы просрочек не было.

Так же отдельно стоит отметить, что чем меньше количество данных, тем менее на них можно правильно полагаться для утверждения результатов. Тем не менее, остальные значения лежат в более или менее одной области (от 7,54 до 9,76%, что соответствует 8,65 +- 1,11%). Данное значение лежит в пределах 13% отклонения от среднего, что является нормальным для данного количества данных. Однако, значение 13% отклонения может быть высоким для банковской сферы и стоит уточнить у заказчика, что является приемлемым. Это поможет сделать правильные выводы для конкретной сферы.

**Вывод:** В ходе анализа данных было выявлено, что при данном количестве данных количество детей не влияет на частоту просрочек по кредиту.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

По аналогии с предыдущим заданием, проведем группировку столбца 'family_status_id' по столбцу 'debt'. Так как в столбце с фактом просрочки по кредитам есть всего 2 значения (0 и 1), то работать можно с ними.

In [25]:
family_debt = data.groupby('family_status_id')['debt'].count() #общее количество данных по статусу семейного положения
print(family_debt)
family_debt_delay = data.groupby('family_status_id')['debt'].sum()  #количетво просрочек по кредиту по статусу семейного положения
family_debt_delay

family_status_id
0    12261
1     4134
2      951
3     1189
4     2796
Name: debt, dtype: int64


family_status_id
0    927
1    385
2     63
3     84
4    273
Name: debt, dtype: int64

Рассчитаем долю просрочек по каждой группе:

In [26]:
family_debt_delay_percent = family_debt_delay / family_debt
family_debt_delay_percent

family_status_id
0    0.075606
1    0.093130
2    0.066246
3    0.070648
4    0.097639
Name: debt, dtype: float64

Итак, доля просрочек по кредитам лежит в диапазоне 6,56-9,74%. Количество измерений для 2х наиболее низких значений доли просрочек в несколько раз ниже остальных, поэтому к ним стоит присмотреться. Вносится предложение увеличить количество данных по этим позициям и запросить их у банка. По данному количеству анализируемых данных, можно говорить о влиянии семейного статуса на вероятность просрочек по выплате кредитов. 

Так же исходя из предоставленных данных можно сделать вввод, что вдовцы/вдовы и категория людей в разводе допускает меньшее количество просрочек по выданным кредитам.

**Вывод:** 

Между семейным положением и возвратом кредита в срок есть корреляция.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Группируем значения ранее выделенного столбца 'total_income_category' по данным в 'debt'.

In [27]:
total_debt = data.groupby('total_income_category')['debt'].count() #общее количество данных по категориям уровня дохода
print(total_debt)
total_debt_delay = data.groupby('total_income_category')['debt'].sum()  #количество просрочек по кредиту по категориям уровня дохода
total_debt_delay

total_income_category
A       25
B     5014
C    15921
D      349
E       22
Name: debt, dtype: int64


total_income_category
A       2
B     354
C    1353
D      21
E       2
Name: debt, dtype: int64

Рассчитаем долю просрочек по категориям дохода:

In [28]:
total_debt_delay_percent = total_debt_delay / total_debt
total_debt_delay_percent

total_income_category
A    0.080000
B    0.070602
C    0.084982
D    0.060172
E    0.090909
Name: debt, dtype: float64

Количество данных по категориям А и Е сильно уступает количеству данных в остальных категориях. Это может быть связано с тем, что людям с низким доходом реже выдают кредиты, а люди с высоким доходом реже в них нуждаются. Тем не менее, разброс между получившимися значениями высок и можно сделать вывод, что доход заемщика влияет на возврат вредита в срок.

**Вывод:** 

Доход заемщика влияет на возврат кредита в срок. Лучше всего со сроками получается у людей с доходом категории 30001–50000 — 'D'.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Сгруппируем данные в столбце 'purpose_category' по столбцу 'debt':

In [29]:
purpose_debt = data.groupby('purpose_category')['debt'].count() #общее количество данных по категориям уровня дохода
print(purpose_debt)
purpose_debt_delay = data.groupby('purpose_category')['debt'].sum()  #количество просрочек по кредиту по категориям уровня дохода
purpose_debt_delay

purpose_category
операции с автомобилем       4279
операции с недвижимостью    10751
получение образования        3988
проведение свадьбы           2313
Name: debt, dtype: int64


purpose_category
операции с автомобилем      400
операции с недвижимостью    780
получение образования       369
проведение свадьбы          183
Name: debt, dtype: int64

Рассчитываем долю имевших просрочки от общего количества по категориям:

In [30]:
purpose_debt_delay_percent = purpose_debt_delay / purpose_debt
purpose_debt_delay_percent

purpose_category
операции с автомобилем      0.093480
операции с недвижимостью    0.072551
получение образования       0.092528
проведение свадьбы          0.079118
Name: debt, dtype: float64

Доля имевших просрочки по кредиту лежит в пределах 7,25-9,35%. Отклонение от среднего значения (8,3 +- 1,05) составляет 12,7%, что для банковской сферы может быть критичным, поэтому стоит уточнить этот вопрос у заказчика. Поверхностная же оценка на данный момент позволяет сделать вывод о низком влиянии цели кредитования на возврат кредита в срок.

**Вывод:**  цель кредитования влияет на возврат кредита в срок.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* В данном случае, пропуски значений были в столбцах о дохлде и о трудовом стаже. Вероятно, эти пропуски могли придти в таблицу в связи с выдачей кредитов "без подтверждения доходов". Так же вероятны случайные пропуски.

<div class="alert alert-danger">
<b>Необходимо исправить ❌</b>

**выдачей кредитов "без подтверждения доходов"** - знаешь такие банки? Сбер? ВТБ24? Райффайзенбанк? Россельхозбанк?    

**Так же вероятны случайные пропуски** - вероятны. Всё может быть конечно же. А сколько таких ошибок в наших данных?
    
Нужно докрутить ответ на этот вопрос. **СОВЕТ:** попробуй посмотреть на **количество пропусков** в наших данных, а также на то, **в каких столбцах** они располагаются, а также на то, **в одних и тех же** строках имеются ли пропуски.
</div>

Поскольку я не  совсем поняла первую часть вашего комментария, то напишу так, как поняла. В таблице не указано, за какое время произведен сбор данных и какие года в нем участвуют. На основании нынешних реалий, законодательства и происходящего в выдаче кредитов в банковской сфере, с уверенностью могу сказать, что все крупные банки выдают кредиты (в том числе, ипотечные средства без подтверждения дохода). И это точно один из крупных источников пропуска данных в столбце доходов.

Стоит отметить, что наличие пропусков в столбце доходов не зависит от вида занятости, цели кредита, семейного положения и других факторов. Но пропуски в столбце дохода совпадают с пропусками в столбце трудового стажа. Исходя из всего вышесказанного, я повторяю свой вывод-эти кредиты, вероятнее всего, были выданы людям без подтверждения дохода. И, расширяя эту мысль, я бы рекомендовала не усреднять значения и не заполнять пропуски средним, а провести анализ для этой категории как для выдачи кредитов без подтверждения дохода и проанализировала бы данные иначе.


<div class="alert alert-success">
<b>КОММЕНТАРИЙ V2</b>

**Поскольку я не совсем поняла первую часть вашего комментария, то напишу так, как поняла. В таблице не указано, за какое время произведен сбор данных и какие года в нем участвуют. На основании нынешних реалий, законодательства и происходящего в выдаче кредитов в банковской сфере, с уверенностью могу сказать, что все крупные банки выдают кредиты (в том числе, ипотечные средства без подтверждения дохода)** - Елена, я зарплатный клиент Райффайзен банка. Поэтому Райфф мне "верит" и кредитную карту мне дали без справок. Но вот другие банки мне не верят, увы )))) Я собираюсь в 2023 взять ипотеку, поэтому когда я через Агентство Ипотечного кредитования подавал заявку, то справка 2-НДФЛ за 2 крайних года - был первым документом в списке.
    
Далее, пропуски в данных есть у людей, которые не работают. Сомнительно всё-таки, что современные банки так великодушны. 
    
В наших данных пропуски в столбцах "трудовой стаж" и "ежемесячный доход" абсолютно идентичны. Количество строк с отсутствующими данными в обоих указанных столбцах совпадает и равно 2174. Маловероятно, что специалисты по выдаче кредитов, заполнив все без исключения ячейки других столбцов по каждому кредитополучателю, оставляли незаполненными именно два показателя, и сделали это ровно 2174 раза в каждом столбце. Даже не проводя проверку, во всех ли случаях незаполненные ячейки находятся в одной и то же строке с высокой долей вероятности можно утверждать, что произошла техническая ошибка при формировании таблицы с исходными данными (либо некий сбой во время выгрузки данных на анализ, либо при объединении данных в единый файл, который также должен был передаваться на анализ, либо эти данные не собирались). Это моё ИМХО (врачи говорят – «второе мнение»).
</div>

Очень хочу акцентировать ответ на этот вопрос. Я тут немного слукавила, делая предположение, поскольку я точно знаю, что кредиты выдаются без подтверждения доходов. И доход, как и стаж, не входят в этих случаях в отчеты. 

На основании данных сайта banki.ru, доля кредитов, выданных крупными банками без подтверждения доходов - около 50%, а микрофинансовых организаций - до 90%.
И да, я в числе официально безработных с ипотекой без подтверждения дохода от Сбербанка.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Медианными значениями лучше заполнять пропуски, тк в данных может быть отклонение от средних значений в ту или иную сторону (аномалии), от которых нужно избавляться, но они могут попасть в расчеты среднего.

### Шаг 4: общий вывод.

Для того, что бы делать вывод о влиянии разных аспектов на возврат кредита в срок, рекоментуется увеличить выборку, на основании которой ведется анализ данных. В некоторых случаях рекомендовано делать выборку по значениям более однородной по количеству измерений (например, уменьшать количество данных до примерно одинакового количества). Так же следует задать вопросы заказчику, какие отклонения от среднего и какие значения в его понимании являются значимыми. 
На данном этапе можно говорить, что некоторые показатели больше влияют на возврат кредита в срок. Наиболее точный вывод с обоснованиями можно сделать на основании беседы с заказчиком.

Далее буду считать, что был проведен разговор с заказчиком, который мне сообщил, что любая разница в доли просрочек является значимой.

***Вывод*** 
В ходе исследования были обработаны данные по ранее выданным кредитам. В ходе исследования было необходимо ответить на поставленные заказчиком воппрсы.

#### Есть ли зависимость между количеством детей и возвратом кредита в срок?
Между количеством детей в семье и возвратом кредита в срок есть связь. Меньше всего просрочек наблюдается в случае, если у заемщика нет детей (просрочки наблюдаются примерно в 7,5% случаев):
0    7,54%
1    9,23%
2    9,45%
3    8,18%
4    9,75%
5    0,00%
При наличии 5 детей в результате анализа было вычислено 0% просрочек, но это значение связано с недостаточным количеством данных (всего 9 кредитов).

#### Есть ли зависимость между семейным положением и возвратом кредита в срок?
В данном случае можно сказать, что вдовцы/вдовы и категория людей в разводе допускает меньшее количество просрочек по выданным кредитам (6,02 и 7,06% соответственно).

#### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?
В данном случае однозначно можно сказать, что меньше всего просрочек по кредитам у заемщиков с уровнем дохода 30001–50000. Это может быть связано с тем, что людям с совсем низким доходам часто просто нечем платить, а те, у кого уровень дохода высок, не всегла рассчитывают свои расходы.

0–30000 — 9,09%
30001–50000 — 6,02%
50001–200000 — 8,50%
200001–1000000 — 7,06%
1000001 и выше — 8,00%

#### Как разные цели кредита влияют на его возврат в срок?
Чаще всего просрочки бывают у людей, которые берут кредит на авто (9,35%) и получают образование(9,25%). Это может быть связано с тем, что машины люди покупают более дорогие, чем могут себе позволить, а кредиты на образование берут студенты или те, кто боятся работу потерять, поэтому переучиваются. 
Реже всего просрочки допускают при кредитовании с целью операций с недвижимостью (7,26%).


### Гипотеза . 
Отдельное исследование кредитов без подтверждения дохода могут изменить результаты исследования.

И я с этим не согласна, тк фактическое распределение в выдаче кредитов будет иным. Эта гипотеза основана только лишь на частоте просрочек по исходным данным и выведенным в результате анализа. Более того, если бы этот вопрс был бы задан напрямую и нужно было бы привести рассчеты по конкретным портретам, вывод был бы иной.